src/aspnetcore/src/Installers/Windows/GenerateNugetPackageWithMsi.ps1

```diff
@@ -16,22 +16,11 @@ param(
     [Parameter(Mandatory=$true)][string]$PackageLicenseExpression
 )
 
-$NuGetDir = Join-Path $RepoRoot "artifacts\Tools\nuget\$Name\$Architecture"
-$NuGetExe = Join-Path $NuGetDir "nuget.exe"
+$NuspecProperties = "ASPNETCORE_RUNTIME_MSI=$MsiPath;ARCH=$Architecture;MAJOR=$MajorVersion;MINOR=$MinorVersion;PackageIcon=$PackageIcon;PackageIconFullPath=$PackageIconFullPath;PackageLicenseExpression=$PackageLicenseExpression"
 
-if (-not (Test-Path $NuGetDir)) {
-    New-Item -ItemType Directory -Force -Path $NuGetDir | Out-Null
-}
+& dotnet pack "$NuspecFile" `
```

Comment:
Have callers pass down the path to the dotnet.exe to use (in particular, pass down the path represented by `$(DotNetTool)` in an MSBuild context (provided by the Arcade SDK so you don't need to define it).